test/tools/jsontests/TransactionTests.cpp

```diff
@@ -38,134 +39,174 @@ namespace fs = boost::filesystem;
 		{
-			string testname = i.first;
-			json_spirit::mObject& o = i.second.get_obj();
+			string const& testname = i.first;
+			json_spirit::mObject const& o = i.second.get_obj();
 
 			if (!TestOutputHelper::get().checkTest(testname))
-			{
-				o.clear(); //don't add irrelevant tests to the final file when filling
 				continue;
-			}
-
-			BOOST_REQUIRE(o.count("blocknumber") > 0);
-			u256 transactionBlock = toInt(o["blocknumber"].get_str());
-			BlockHeader bh;
-			bh.setNumber(transactionBlock);
-			bh.setGasLimit(u256("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
-			bool onConstantinople = (transactionBlock >= se->chainParams().constantinopleForkBlock);
 
 			if (_fillin)
-			{
-				BOOST_REQUIRE_MESSAGE(o.count("transaction") > 0, "transaction section not found! " + TestOutputHelper::get().testFileName());
-				mObject tObj = o["transaction"].get_obj();
-
-				//Construct Rlp of the given transaction
-				RLPStream rlpStream = createRLPStreamFromTransactionFields(tObj);
-				o["rlp"] = toHexPrefixed(rlpStream.out());
-
-				try
-				{
-					Transaction txFromFields(rlpStream.out(), CheckTransaction::Everything);
-					bool onConstantinopleAndZeroSig = onConstantinople && txFromFields.hasZeroSignature();
-
-					if (!txFromFields.signature().isValid())
-					if (!onConstantinopleAndZeroSig)
-						BOOST_THROW_EXCEPTION(Exception() << errinfo_comment(testname + "transaction from RLP signature is invalid") );
-					se->verifyTransaction(ImportRequirements::Everything, txFromFields, bh, 0);
-
-					if (o.count("sender") > 0)
-					{
-						string expectSender = toString(o["sender"].get_str());
-						BOOST_CHECK_MESSAGE(toString(txFromFields.sender()) == expectSender, "Error filling transaction test " + TestOutputHelper::get().testName() + ": expected another sender address! (got: " + toString(txFromFields.sender()) + "), expected: (" + expectSender + ")");
-					}
-					o["sender"] = toString(txFromFields.sender());
-					o["transaction"] = ImportTest::makeAllFieldsHex(tObj);
-					o["hash"] = toString(txFromFields.sha3());
-				}
-				catch(Exception const& _e)
-				{
-					//Transaction is InValid
-					cnote << "Transaction Exception: " << diagnostic_information(_e);
-					o.erase(o.find("transaction"));
-					if (o.count("sender") > 0)
-						o.erase(o.find("sender"));
-					if (o.count("expect") > 0)
-					{
-						bool expectInValid = (o["expect"].get_str() == "invalid");
-						BOOST_CHECK_MESSAGE(expectInValid, testname + " Check state: Transaction '" << i.first << "' is expected to be valid!");
-						o.erase(o.find("expect"));
-					}
-				}
-
-				//Transaction is Valid
-				if (o.count("expect") > 0)
-				{
-					bool expectValid = (o["expect"].get_str() == "valid");
-					BOOST_CHECK_MESSAGE(expectValid, testname + " Check state: Transaction '" << i.first << "' is expected to be invalid!");
-					o.erase(o.find("expect"));
-				}
-			}
+				v[testname] = FillTransactionTest(o);
```

Comment:
Maybe also call `TestTransactionTest` on `v[testname]`?

Reply:
it is done from the TestSuite